src/main/java/com/orbitz/consul/SessionClient.java

```diff
@@ -8,176 +10,202 @@
-      }
-
-      Response session = target.path("renew").path(sessionId).request().put(Entity.entity("{}",
-              MediaType.APPLICATION_JSON_TYPE));
-      boolean result = session.hasEntity();
-      session.close();
-      return result;
-
-   }
-
-   /**
-    * Destroy session.
-    *
-    * PUT /v1/session/destroy/{sessionId}
-    *
-    * @param sessionId
-    *
-    * @return ID of the newly created session .
-    */
-   public boolean destroySession(final String sessionId) {
-      return destroySession(sessionId, null);
-   }
-
-   /**
-    * Destroy session.
-    *
-    * PUT /v1/session/destroy/{sessionId}
-    *
-    * @param sessionId
-    * @param dc Data center
-    *
-    * @return ID of the newly created session .
-    */
-   public boolean destroySession(final String sessionId, final String dc) {
-      WebTarget target = webTarget;
-
-      if (dc != null) {
-         target = webTarget.queryParam("dc", dc);
-      }
-
-      return target.path("destroy").path(sessionId).request().put(Entity.entity("",
-              MediaType.TEXT_PLAIN_TYPE), Boolean.class);
-   }
-
-   /**
-    * Retrieves session info.
-    *
-    * GET /v1/session/info/{sessionId}
-    *
-    * @param sessionId
-    *
-    * @return {@link SessionInfo}.
-    */
-   public Optional<SessionInfo> getSessionInfo(final String sessionId) {
-      return getSessionInfo(sessionId, null);
-   }
-
-   /**
-    * Retrieves session info.
-    *
-    * GET /v1/session/info/{sessionId}
-    *
-    * @param sessionId
-    * @param dc Data center
-    *
-    * @return {@link SessionInfo}.
-    */
-   public Optional<SessionInfo> getSessionInfo(final String sessionId, final String dc) {
-      WebTarget target = webTarget;
-
-      if (dc != null) {
-         target = target.queryParam("dc", dc);
-      }
-
-      target = addParams(target.path("info").path(sessionId), QueryOptions.BLANK);
-
-      List<SessionInfo> sessionInfo = Arrays.asList(target
-              .request().accept(MediaType.APPLICATION_JSON_TYPE).get(SessionInfo[].class));
-
-      return sessionInfo != null && sessionInfo.size() != 0 ? Optional.of(sessionInfo.get(0)) : Optional.<SessionInfo>absent();
-   }
-
+    private static final GenericType<SessionCreatedResponse> SESSION_CREATED_RESPONSE_TYPE =
+            new GenericType<SessionCreatedResponse>() {
+            };
+    private static final GenericType<List<SessionInfo>> SESSION_INFO_LIST_TYPE =
+            new GenericType<List<SessionInfo>>() {
+            };
+
+    private final WebTarget webTarget;
+
+    /**
+     * Constructs an instance of this class.
+     *
+     * @param webTarget The {@link WebTarget} to base requests from.
+     */
+    SessionClient(WebTarget webTarget) {
+        this.webTarget = webTarget;
+    }
+
+    /**
+     * Retrieves the host/port of the Consul leader.
+     * 
+     * GET /v1/status/leader
+     *
+     * @return The host/port of the leader.
+     */
+    public String getLeader() {
```

Comment:
This doesn't seem like it really belongs here since the same method exists in `StatusClient`